nmdc_runtime/api/db/mongo.py

```diff
@@ -73,3 +78,107 @@ def activity_collection_names(mdb: MongoDatabase) -> Set[str]:
         "functional_annotation_set",
         "genome_feature_set",
     }
+
+
+def all_docs_have_unique_id(coll) -> bool:
+    first_doc = coll.find_one({}, ["id"])
+    if first_doc is None or "id" not in first_doc:
+        # short-circuit exit for empty collection or large collection via first-doc peek.
+        return False
+
+    total_count = coll.count_documents({})
+    return len(coll.distinct("id")) == total_count
+
+
+def ensure_unique_id_indexes(mdb: MongoDatabase):
+    """Ensure that any collections with an "id" field have an index on "id"."""
+    candidate_names = (
+        set(mdb.list_collection_names()) | schema_collection_names_with_id_field()
+    )
+    for collection_name in candidate_names:
+        if collection_name.startswith("system."):  # reserved by mongodb
```

Comment:
Should we also ignore "internal" NMDC collections beginning with _

Reply:
Interesting. I'd like to not ignore them, and lean on the semantic that a collection with at least one document, an "id" field in every document, and all "id" values different, i.e. what is checked by `all_docs_have_unique_id`, should ensure a unique index on "id".

I can be swayed to also ignore "_"-beginning collection names if you want that to approve this PR. The current behavior is inessential.